atc/exec/retry_error_step.go

```diff
@@ -0,0 +1,53 @@
+package exec
+
+import (
+	"context"
+	"reflect"
+	"regexp"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+)
+
+type Retriable struct{}
+
+func (r Retriable) Error() string {
+	return "retriable"
+}
+
+type RetryErrorStep struct {
+	Step
+
+	delegate LogErrorStepDelegate
+}
+
+func RetryError(step Step, delegate LogErrorStepDelegate) Step {
+	return RetryErrorStep{
+		Step:     LogError(step, delegate),
+		delegate: delegate,
+	}
+}
+
+func (step RetryErrorStep) Run(ctx context.Context, state RunState) error {
```

Comment:
Is there a way to unit test this functionality? You can look at any of the other step files for test examples. Here's one:
https://github.com/concourse/concourse/blob/master/atc/exec/ensure_step_test.go

Reply:
I'll add unit test for this. Practically, for this kind of new feature changes, I usually called for an initial review first from design perspective, if you folks agree with the design, I added tests in the second commit. Because adding tests is kinda costly, if we added folks rejected the design, the time spent on writing unit test is wasted. I have done many PRs, and always did the way.